src/react-components/profile-entry-panel.js

```diff
@@ -94,20 +96,46 @@ class ProfileEntryPanel extends Component {
               title={formatMessage({ id: "profile.display_name.validation_warning" })}
               ref={inp => (this.nameInput = inp)}
             />
-            <div className={styles.avatarSelectorContainer}>
-              <div className="loading-panel">
-                <div className="loader-wrap">
-                  <div className="loader">
-                    <div className="loader-center" />
+            {this.state.customMode ? (
+              <div className={styles.avatarSelectorContainer}>
+                <label htmlFor="#custom-avatar-url" className={styles.title}>
+                  Avatar GLTF/GLB{" "}
+                </label>
+                <input
+                  id="custom-avatar-url"
+                  type="url"
+                  className={styles.formFieldText}
+                  value={this.state.avatarId}
+                  onChange={e => this.setState({ avatarId: e.target.value })}
+                />
+                <div className={styles.links}>
+                  <a onClick={() => this.setState({ customMode: false, avatarId: "botdefault" })}>cancel</a>
+                </div>
+              </div>
+            ) : (
+              <div className={styles.avatarSelectorContainer}>
+                <div className="loading-panel">
+                  <div className="loader-wrap">
+                    <div className="loader">
+                      <div className="loader-center" />
+                    </div>
                   </div>
                 </div>
+                <iframe
+                  className={styles.avatarSelector}
+                  src={`/avatar-selector.html#avatar_id=${this.state.avatarId}`}
+                  ref={ifr => (this.avatarSelector = ifr)}
+                />
+                <a
+                  className="custom-url-link"
+                  onClick={() =>
+                    this.setState({ customMode: true, avatarId: avatars.find(a => a.id === this.state.avatarId).model })
+                  }
+                >
+                  options
```

Comment:
I would actually prefer if this said "custom url" for now, since most people will click on an "options" link expecting to find generally useful options, whereas this url field will probably be niche.